src/types/notification.ts

```diff
@@ -21,3 +24,83 @@ export const Notification = t.interface({
 });
 
 export type Notification = t.TypeOf<typeof Notification>;
+
+/**
+ * Type of device.
+ */
+export enum DevicePlatformEnum {
+  apns = "apns",
+  gcm = "gcm"
+}
+
+export type DevicePlatform = t.TypeOf<typeof DevicePlatform>;
+
+export const DevicePlatform = enumType<DevicePlatformEnum>(
+  DevicePlatformEnum,
+  "DevicePlatform"
+);
+
+/**
+ * An installation ID.
+ */
+export type InstallationID = t.TypeOf<typeof InstallationID>;
+
+export const InstallationID = PatternString(
+  "^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$"
+);
+
+/**
+ * Device data.
+ */
+export const Device = t.interface({
+  platform: DevicePlatform,
+  pushChannel: PatternString("[0-9a-fA-F]{64}")
+});
+
+export type Device = t.TypeOf<typeof Device>;
+
+/**
+ * An hashed fiscal code.
+ */
+interface IFiscalCodeHashTag {
+  readonly kind: "IFiscalCodeHashTag";
+}
+
+export const FiscalCodeHash = tag<IFiscalCodeHashTag>()(NonEmptyString);
+
+export type FiscalCodeHash = t.TypeOf<typeof FiscalCodeHash>;
+
+/**
+ * Notification template.
+ */
+export interface INotificationTemplate {
+  readonly body: string;
```

Comment:
ok I see this is also specific to Notification hubs, pls add links to the relevant docs so that when reading this code one can understand why things are made in this way